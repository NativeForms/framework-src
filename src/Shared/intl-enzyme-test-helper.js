/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies

/** Create the IntlProvider to retrieve context for wrapping around. */
function createIntlContext(messages, locale) {
  const intlProvider = new IntlProvider({ messages, locale }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
}

/** When using React-Intl `injectIntl` on components, props.intl is required. */
function nodeWithIntlProp(node, messages = {}, locale = 'en') {
  return React.cloneElement(node, { intl: createIntlContext(messages, locale) });
}

/**
 * Create a shadow renderer that wraps a node with Intl provider context.
 * @param {ReactComponent} node - Any React Component
 * @param {Object} context
 * @param {Object} messages - A map with keys (id) and messages (value)
 * @param {string} locale - Locale string
 */
export function shallowWithIntl(node, { context } = {}, messages = {}, locale = 'en') {
  return shallow(
    nodeWithIntlProp(node),
    {
      context: Object.assign({}, context, { intl: createIntlContext(messages, locale) }),
    }
  );
}

/**
 * Mount the node with Intl provider context.
 * @param {Component} node - Any React Component
 * @param {Object} context
 * @param {Object} messages - A map with keys (id) and messages (value)
 * @param {string} locale - Locale string
 */
export function mountWithIntl(node, { context, childContextTypes } = {}, messages = {}, locale = 'en') {
  return mount(
    nodeWithIntlProp(node),
    {
      context: Object.assign({}, context, { intl: createIntlContext(messages, locale) }),
      childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes)
    }
  );
}
