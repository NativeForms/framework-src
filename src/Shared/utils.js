import React from 'react';
import syntaxHighlighter from 'react-native-syntax-highlighter';
import { github } from 'react-syntax-highlighter/dist/styles';
import { Iterable } from 'immutable';

export const SchemaViewer = (props) => syntaxHighlighter(
  Object.assign({}, syntaxHighlighter.defaultProps, {
    language: 'javascript',
    style: github,
    codeTagProps: {
      alwaysBounceVertical: false
    }
  }, props));

// See http://redux.js.org/docs/recipes/UsingImmutableJS.html#use-a-higher-order-component-to-convert-your-smart-components-immutablejs-props-to-your-dumb-components-javascript-props
export const toJS = (WrappedComponent) =>
  (wrappedComponentProps) => {
    const KEY = 0;
    const VALUE = 1;

    const propsJS = Object.entries(wrappedComponentProps)
        .reduce((newProps, wrappedComponentProp) => {
          /* eslint no-param-reassign:0 */
          newProps[wrappedComponentProp[KEY]] =
              Iterable.isIterable(wrappedComponentProp[VALUE])
                  ? wrappedComponentProp[VALUE].toJS()
                  : wrappedComponentProp[VALUE];
          return newProps;
        }, {});

    return <WrappedComponent {...propsJS} />;
  };

export function getLocale() {
  return 'en';
}
