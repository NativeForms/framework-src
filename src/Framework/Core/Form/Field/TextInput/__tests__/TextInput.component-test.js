import React from 'react';
import { intlShape } from 'react-intl';

// framework
import TextInputComponent from '../TextInput.component';
import i18n from '../TextInput.i18n';
import schema from '../TextInput.schema';

// shared 
import { shallowWithIntl } from '../../../../../../Shared/intl-enzyme-test-helper';

const message = i18n.en;
const props = schema.attributes.formComponents;


describe('components <TextInputComponent />', () => {
  let wrapper;
  props.forEach((prop) => {
    it('renders text input component correctly', () => {
      wrapper = shallowWithIntl(
        <TextInputComponent {...prop.attributes} />, {}, message
    );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
