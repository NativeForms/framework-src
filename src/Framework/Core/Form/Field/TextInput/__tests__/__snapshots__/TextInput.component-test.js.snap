// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <TextInputComponent /> renders text input component correctly 1`] = `
<TextInputComponent
  clearButton={false}
  defaultText={null}
  headerLabel={null}
  hiddenText={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "defaultText": "Default Text",
        "footDescription": "Foot Description",
        "headerLabel": "Header Label",
        "label": "Label",
        "title": "Text Input",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label="Key"
  multiline={false}
  value={null}
/>
`;

exports[`components <TextInputComponent /> renders text input component correctly 2`] = `
<TextInputComponent
  clearButton={false}
  defaultText={null}
  headerLabel={null}
  hiddenText={true}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "defaultText": "Default Text",
        "footDescription": "Foot Description",
        "headerLabel": "Header Label",
        "label": "Label",
        "title": "Text Input",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label="Secret"
  multiline={false}
  value={null}
/>
`;

exports[`components <TextInputComponent /> renders text input component correctly 3`] = `
<TextInputComponent
  clearButton={true}
  defaultText={null}
  headerLabel={null}
  hiddenText={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "defaultText": "Default Text",
        "footDescription": "Foot Description",
        "headerLabel": "Header Label",
        "label": "Label",
        "title": "Text Input",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label="Value 1"
  multiline={false}
  value="Delete Me"
/>
`;

exports[`components <TextInputComponent /> renders text input component correctly 4`] = `
<TextInputComponent
  clearButton={false}
  defaultText="Replace Me"
  headerLabel={null}
  hiddenText={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "defaultText": "Default Text",
        "footDescription": "Foot Description",
        "headerLabel": "Header Label",
        "label": "Label",
        "title": "Text Input",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label="Value 2"
  multiline={false}
  value={null}
/>
`;

exports[`components <TextInputComponent /> renders text input component correctly 5`] = `
<TextInputComponent
  clearButton={false}
  defaultText={null}
  headerLabel="Header Label"
  hiddenText={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "defaultText": "Default Text",
        "footDescription": "Foot Description",
        "headerLabel": "Header Label",
        "label": "Label",
        "title": "Text Input",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label={null}
  multiline={false}
  value={null}
/>
`;

exports[`components <TextInputComponent /> renders text input component correctly 6`] = `
<TextInputComponent
  clearButton={false}
  defaultText={null}
  headerLabel="Description"
  hiddenText={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "defaultText": "Default Text",
        "footDescription": "Foot Description",
        "headerLabel": "Header Label",
        "label": "Label",
        "title": "Text Input",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label={null}
  multiline={true}
  value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Itaque hic ipse iam pridem est reiectus; De ingenio eius in his disputationibus, non de moribus quaeritur. Nunc haec primum fortasse audientis servire debemus. Atqui iste locus est, Piso, tibi etiam atque etiam confirmandus, inquam;"
/>
`;
